### PR TITLE
fuzz: version handshake

### PR DESCRIPTION
**Pull request by maflcko**
Not fuzzing the version handshake will limit fuzz coverage

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

**Comment by practicalswift** [comment]
Tested ACK dd8654bcd53dade1f5114d68bb51ef22e585b9ba

File: src/test/fuzz/process_messages.cpp [hunk 1]
```diff
@@ -48,14 +48,16 @@ void test_one_input(const std::vector<uint8_t>& buffer)
...
+        const ConnectionType conn_type = fuzzed_data_provider.PickValueInArray(ALL_CONNECTION_TYPES);
         peers.push_back(MakeUnique<CNode>(i, service_flags, 0, INVALID_SOCKET, CAddress{CService{in_addr{0x0100007f}, 7777}, NODE_NETWORK}, 0, 0, CAddress{}, std::string{}, conn_type).release());
         CNode& p2p_node = *peers.back();
 
+        connman.SetPeerPermissionFlags(p2p_node, permission_flags);
         p2p_node.fSuccessfullyConnected = true;
-        p2p_node.fPauseSend = false;
         p2p_node.nVersion = PROTOCOL_VERSION;
         p2p_node.SetCommonVersion(PROTOCOL_VERSION);
+        p2p_node.fPauseSend = false;
```
Comment (Crypt-iQ) [inline src/test/fuzz/process_messages.cpp:60]: Why was this moved?

**Review by Crypt-iQ — COMMENTED** [review]
utACK dd8654bcd53dade1f5114d68bb51ef22e585b9ba

Running now to compare coverage before and after this patch. I like that the fuzzers now have `SendMessages` coverage.

File: src/test/fuzz/process_messages.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/test/fuzz/process_messages.cpp:60]: There are some more changes that I haven't yet pushed to increase coverage even further. They change the seed format as well, so I am going to mark this as draft for now.

Comment (Crypt-iQ) [reply src/test/fuzz/process_messages.cpp:60]: What is the procedure for when a PR needs to modify the fuzzing harness and may invalidate all of the seeds? Should something akin to a database migration be used if possible, applied to the seeds to get at least the same coverage as before?

Comment (MarcoFalke) [reply src/test/fuzz/process_messages.cpp:60]: The format might be arbitrarily complex (even depending on the consensus rules or other Bitcoin Core code), so I don't think this can be done automatically in a trivial manner. I mostly do it manually if the fuzz engine can't figure it out by itself after a bit of CPU time.

**Comment by Crypt-iQ** [comment]
Coverage from dd8654bcd53dade1f5114d68bb51ef22e585b9ba:
https://crypt-iq.github.io/20370_review/covrun1/src/net_processing.cpp.gcov.html

Compared to a recent `master` commit (should have recorded this...oh well):
https://crypt-iq.github.io/20370_review/covrun_master/src/net_processing.cpp.gcov.html

Coverage goes up because of `SendMessages`, but `ProcessMessages` coverage is worse.  I think it's probably due to the seed change.

**Comment by MarcoFalke** [comment]
Rebased

**Comment by MarcoFalke** [comment]
Rebased

**Comment by practicalswift** [comment]
cr ACK fabce459bb44e90dc7ae9c44eeedab707435af5b: patch looks very much correct
